doc/style-guide-javascript.md

```diff
@@ -0,0 +1,185 @@
+        * When documenting a `Promise`, indicate the value that will be provided if the promise is resolved.
+    * Multiple line comments end with `*/` and have leading `*` on each line.  The stars all line up throughout the comment.
+    * Wrap comments at 80 characters except when using a code block as an example in the JSDoc.
+    * Do not comment code unless it's an example.  We're using a version control system for a reason.
+    * Comment any tricky bits of code or rewrite them to be not tricky.
+* Indentation and Spacing
+    * Newlines after open braces and open brackets (both `{` and `[`) and increase indent level.
+    * Newlines before and decrease indent level with closing braces and brackets (both `}` and `]`).
+    * Spaces before and after most operators (`=`, `+`, etc.).
+    * Single line comments on a line of their own should have a newline above them.
+    * Multi-line comments will be on their own line and have a newline above them.
+    * New lines should separate any control structures (`if`, `while`, `for`, `switch`) from other code.  They should not be separated if the next/previous line is a brace.
+    * New lines after commas in object literals and array literals.
+    * There shouldn't be any other newlines.
+    * Spaces before `{`, `[`, `(` unless preceded by another one of those symbols.  Same goes for a space after the closing `}`, `]` and `)`.  Do not add a space before `(` when calling a function or defining a named function.
+    * ESLint forces `case` statements and the `default` case in switches to be un-indented (just that line).  There must be a blank line after `break;` as well.
+* Inline Objects and Arrays, Defining Objects
+    * No trailing comma after the last property or element.
+    * Property names shall be in alphabetical order, followed immediately by a colon, a space, and the property's value.
+    * Property names should be unquoted unless it is necessary to wrap it in quotes.
+    * When defining a class, its methods are all alphabetized.
+* Order of items within a function
+    * ESLint flags (only if necessary).
+    * `var` declarations:
+        * One declaration only.
+        * Variables will be alphabetized.
+        * No assignments.
+        * One line.
+    * Functions, sorted alphabetically when possible.
+    * Your code.
+* Specific Constructs
+    * Nothing that's displayable to the user should be sourced from JavaScript.  It should come from an HTML template, a JSON file from a server or API calls.
+    * Use double-quoted strings.
+    * We code for ECMA Script 6.
+    * All control structures, like `if` statements, require braces.
+    * Ternary statements should be avoided.
+    * Avoid manual minification - let a minifier do it.
+        * Ternary statements should be expanded into `if` blocks.
+        * While `!!~haystack.indexOf(needle)` is tricky and neat, write `str.indexOf(needle) !== -1` so all programmers on the project immediatly know what the code does.
```

Comment:
Getting pretty nit picky but in one example you use `haystack` and in the next you use `str`